core/controller.go

```diff
@@ -77,7 +80,7 @@ type Controller struct {
 var _ IController = (*Controller)(nil)
 
 // NewController creates a new instance of Controller
-func NewController(spec *config.Specification, fileServer http.Handler) *Controller {
+func NewController(spec *config.Specification, fileServer http.Handler, listen [] config.ListenSetting) *Controller {
```

Comment:
`spec` here should have the `ListenSetting` we don't need to pass it

Reply:
I struggled to access the data for some reason that way which is why I did that, but I can try again.